File: 30 Days of Code/Day 20 Sorting/Solution.py
# https://www.hackerrank.com/challenges/30-sorting/problem

#!/bin/python3

import sys

n = int(input().strip())
a = list(map(int, input().strip().split(' ')))
# Write Your Code Here
numSwaps = 0
for i in range(len(a)):
    temp = 0
    for j in range(len(a)-1):
        if a[j] > a[j+1]:
            a[j] = a[j] + a[j+1]
            a[j+1] = a[j] - a[j+1]
            a[j] = a[j] - a[j+1]
            numSwaps += 1
            temp += 1
    if temp == 0:
        break

print("Array is sorted in {} swaps.".format(numSwaps))
print("First Element: {}".format(a[0]))
print("Last Element: {}".format(a[-1]))

File: 30 Days of Code/Day 20 Sorting/Solution.cpp
// https://www.hackerrank.com/challenges/30-sorting/problem

#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for(int a_i = 0; a_i < n; a_i++){
    	cin >> a[a_i];
    }
    // Write Your Code Here
    int numSwaps = 0;
    for(int i=0; i<a.size(); ++i){
        int temp = 0;
        for(int j=0; j<a.size()-1; ++j){
            if(a[j]>a[j+1]){
                a[j] = a[j] + a[j+1];
                a[j+1] = a[j] - a[j+1];
                a[j] = a[j] - a[j+1];
                ++temp;
                ++numSwaps;
            }
        }
        if(temp==0){
            break;
        }
    }
    cout<<"Array is sorted in "<<numSwaps<<" swaps."<<endl;
    cout<<"First Element: "<<a[0]<<endl;
    cout<<"Last Element: "<<a[a.size()-1]<<endl;
    return 0;
}